src/main/java/com/mongodb/spark/sql/connector/write/MongoDataWriter.java

```diff
@@ -79,7 +79,8 @@ final class MongoDataWriter implements DataWriter<InternalRow> {
             schema, writeConfig.convertJson(), writeConfig.ignoreNullValues());
     this.writeConfig = writeConfig;
     this.epochId = epochId;
-    this.bulkWriteOptions = new BulkWriteOptions().ordered(writeConfig.isOrdered());
+    this.bulkWriteOptions =
+        new BulkWriteOptions().ordered(writeConfig.isOrdered()).comment(writeConfig.getComment());
```

Comment:
[nit] I noticed that in many places, the method chains are wrapped on a separate line, which I personally find easier to read

```suggestion
        new BulkWriteOptions()
        	.ordered(writeConfig.isOrdered())
       		.comment(writeConfig.getComment());
```